src/components/Note.tsx

```diff
@@ -81,7 +84,20 @@ const Note = React.memo(
         else if (e.key === 'Backspace' && !note) {
           e.stopPropagation()
           e.preventDefault()
-          dispatch(deleteAttribute({ path, value: '=note' }))
+
+          // delete target thought if it exists or =note attribute if it exists
+          dispatch((dispatch, getState) => {
+            const state = getState()
+            const targetThoughtPath = resolveNotePath(state, path)
+            const targetThought = targetThoughtPath ? getThoughtById(state, head(targetThoughtPath)) : undefined
+
+            if (targetThought && targetThought.value !== '=note') {
+              dispatch(deleteThought({ pathParent: path, thoughtId: targetThought.id }))
+            } else {
+              dispatch(deleteAttribute({ path, value: '=note' }))
+            }
```

Comment:
Look for a way to combine these cases. There shouldn't be anything fundamentally different about a regular note and a note determined by a target path. Maybe something like this?

```ts
const targetThoughtPath = resolveNotePath(state, path) ?? path
dispatch(deleteThought({ pathParent: path, thoughtId: head(targetThoughtPath) }))
```

This would be even simpler if `resoloveNotePath` didn't return `null` for the plain `=note` case.